main/test/flix/Test.Exp.Jvm.NewObject.flix

```diff
@@ -1,285 +1,313 @@
-      def testMethod(_this: ##flix.test.TestBoolArrayInterface, xs: Array[Bool, false]): Array[Bool, false] = 
+    import static flix.test.TestBoolArrayInterface.runTest(TestBoolArrayInterface): Bool;
+    let anon = object TestBoolArrayInterface {
+      def testMethod(_this: TestBoolArrayInterface, xs: Array[Bool, false]): Array[Bool, false] = 
         xs |> Array.map(x -> not x)
     };
     runTest(anon)
 
   @test
   def testCharArrayInterface01(): Bool & Impure =
-    import static flix.test.TestCharArrayInterface.runTest(##flix.test.TestCharArrayInterface): Bool;
-    let anon = object ##flix.test.TestCharArrayInterface {
-      def testMethod(_this: ##flix.test.TestCharArrayInterface, xs: Array[Char, false]): Array[Char, false] = 
+    import static flix.test.TestCharArrayInterface.runTest(TestCharArrayInterface): Bool;
+    let anon = object TestCharArrayInterface {
+      def testMethod(_this: TestCharArrayInterface, xs: Array[Char, false]): Array[Char, false] = 
         xs |> Array.map(Char.toUpperCase)
     };
     runTest(anon)
 
   @test
   def testInt8ArrayInterface01(): Bool & Impure =
-    import static flix.test.TestInt8ArrayInterface.runTest(##flix.test.TestInt8ArrayInterface): Bool;
-    let anon = object ##flix.test.TestInt8ArrayInterface {
-      def testMethod(_this: ##flix.test.TestInt8ArrayInterface, xs: Array[Int8, false]): Array[Int8, false] = 
+    import static flix.test.TestInt8ArrayInterface.runTest(TestInt8ArrayInterface): Bool;
+    let anon = object TestInt8ArrayInterface {
+      def testMethod(_this: TestInt8ArrayInterface, xs: Array[Int8, false]): Array[Int8, false] = 
         xs |> Array.map(x -> x + 1i8)
     };
     runTest(anon)
 
   @test
   def testInt16ArrayInterface01(): Bool & Impure =
-    import static flix.test.TestInt16ArrayInterface.runTest(##flix.test.TestInt16ArrayInterface): Bool;
-    let anon = object ##flix.test.TestInt16ArrayInterface {
-      def testMethod(_this: ##flix.test.TestInt16ArrayInterface, xs: Array[Int16, false]): Array[Int16, false] = 
+    import static flix.test.TestInt16ArrayInterface.runTest(TestInt16ArrayInterface): Bool;
+    let anon = object TestInt16ArrayInterface {
+      def testMethod(_this: TestInt16ArrayInterface, xs: Array[Int16, false]): Array[Int16, false] = 
         xs |> Array.map(x -> x + 1i16)
     };
     runTest(anon)
 
   @test
   def testInt32ArrayInterface01(): Bool & Impure =
-    import static flix.test.TestInt32ArrayInterface.runTest(##flix.test.TestInt32ArrayInterface): Bool;
-    let anon = object ##flix.test.TestInt32ArrayInterface {
-      def testMethod(_this: ##flix.test.TestInt32ArrayInterface, xs: Array[Int32, false]): Array[Int32, false] = 
+    import static flix.test.TestInt32ArrayInterface.runTest(TestInt32ArrayInterface): Bool;
+    let anon = object TestInt32ArrayInterface {
+      def testMethod(_this: TestInt32ArrayInterface, xs: Array[Int32, false]): Array[Int32, false] = 
```

Comment:
I just saw this now, but the `false` should really be `Static`.

Reply:
I will make the change. But...

If these types are wrong, then the error message currently output with `--explain` currently suggests `Array[Int32, false`):

```
-- Safety Error -------------------------------------------------- main/test/flix/Test.Exp.Jvm.NewObject.flix

>> No implementation found for method 'testMethod' of superclass 'TestInt32ArrayInterface'.

148 |>     let anon = object ##flix.test.TestInt32ArrayInterface {
149 |>       def testMethod(_this: ##flix.test.TestInt32ArrayInterface, xs: Array[Int32, Static]): Array[Int32, _] = 
150 |>         xs |> Array.map(x -> x + 1)
151 |>     };

the object occurs here.

 Try adding a method with the following signature:

 def testMethod(Array[Int32, false]): Array[Int32, false]
```

This is because the string generated by `FormatType.formatWellKindedType` says `false` not `Static`.

Is this a bug? Or..?

Reply:
That's a bug, or rather its because we are in a longer process. In `Prelude.flix` you can see that `Static = False`, but we should use Static everywhere so we can change the definition of it later.